common/web/input-processor/src/text/inputProcessor.ts

```diff
@@ -193,7 +197,7 @@ namespace com.keyman.text {
       this.keyboardProcessor.newLayerStore.set(hasLayerChanged ? this.keyboardProcessor.layerId : '');
       this.keyboardProcessor.oldLayerStore.set(hasLayerChanged ? startingLayerId : '');
 
-      let postRuleBehavior = this.keyboardProcessor.processPostKeystroke(keyEvent.device, outputTarget);
+      let postRuleBehavior = this.keyboardProcessor.processPostKeystroke(this.contextDevice, outputTarget);
```

Comment:
This is the main change here - the one that directly fixes the bug.

Most of the other changes involve a rename of the field used here to better clarify and document its role.